src/patterns/scheduled-ecs-task.ts

```diff
@@ -0,0 +1,239 @@
+ *
+ * const props = {
+ *   // Other props here
+ *   schedule: Schedule.expression("cron(0 8 ? * MON-FRI *)"),
+ * }
+ * ```
+ *
+ * See [[`ContainerConfiguration`]] for details of how to configure the container used to run the task.
+ *
+ * `taskTimeoutInMinutes` does what is says on the tin. The default timeout is 15 minutes.
+ *
+ * The `taskCommand` prop allows you to specify a command to run when the task starts (this can also be done via a CMD statement in
+ * your Dockerfile). For example:
+ *
+ * const props = {
+ *    //other props
+ *    taskCommand: `aws s3 cp s3://${distbucket}/${stack}/${stage}/${app}/task.sh . && ./task.sh
+    }
+ *
+ * It is advisable to configure alarms for when the job fails/times out. To do this specify the alarmSnsTopicArn prop.
+ *
+ * `customTaskPolicies` allows your task to interact with other AWS services. By default a task
+ * will have read access to the distribution bucket for your account.
+ *
+ * You can specify security groups to apply to task using the `securityGroups` prop.
+ *
+ * You can also set the memory and cpu units for your task. See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size
+ * for further details.
+ *
+ */
+export interface GuScheduledEcsTaskProps extends Identity {
+  vpc: IVpc;
+  schedule: Schedule;
+  containerConfiguration: ContainerConfiguration;
+  taskTimeoutInMinutes?: number;
+  cpu?: number;
+  memory?: number;
+  taskCommand?: string;
+  alarmSnsTopicArn?: string;
+  securityGroups?: ISecurityGroup[];
+  customTaskPolicies?: PolicyStatement[];
+}
+
+/**
+ * Containers can be specified either via a repository and version or by container id (in which case it will
+ * be fetched from docker hub).
+ */
+const getContainer = (config: ContainerConfiguration) => {
+  const ecrContainer =
+    config.id && config.repository && ContainerImage.fromEcrRepository(config.repository, config.version);
+
+  if (ecrContainer) {
+    return ecrContainer;
+  } else {
+    return ContainerImage.fromRegistry(config.id ?? "ubuntu:focal");
+  }
+};
+
+/**
+ * Pattern which creates all of the resources needed to invoke a fargate task on a schedule.
+ *
+ * The task will be wrapped in a step function to allow for easier triggering and alarming on failure.
+ *
+ * For all configuration options, see [[`GuScheduledEcsTaskProps`]].
+ *
+ * Note that if your task reliably completes in less than 15 minutes then you should probably use a [[`GuScheduledLambda`]] instead. This
+ * pattern was mainly created to work around the 15 minute lambda timeout.
+ */
+export class GuScheduledEcsTask {
+  constructor(scope: GuStack, props: GuScheduledEcsTaskProps) {
+    const distBucket = GuDistributionBucketParameter.getInstance(scope).valueAsString;
+
+    const timeout = props.taskTimeoutInMinutes ?? 15;
+
+    const cpu = props.cpu ?? 2048;
+    const memory = props.memory ?? 4096;
+
+    const cluster = new Cluster(scope, `${props.app}-cluster`, {
```

Comment:
Could we use the [`AppIdentity.suffixText`](https://github.com/guardian/cdk/blob/d1933e39a5d12395c17f236406a2905f35d04582/src/constructs/core/identity.ts#L16) helper here (and other places where `app` is used for the construct ID) for consistency with the rest of the repository? (Again, we've got some work in the backlog to somewhat automate this away as it's annoying to remember!).